dotnet store fails with "the destination is a folder instead of a file" in 3.0 SDK

This is based on the repro from dotnet/cli#11223 which can be found [here](https://github.com/narcis-pv/dotnet-core-target-manifest-bug).

Running the `dotnet store` command described in the repro with .NET Core SDK 3.0.100-preview6-011540 produces error like this:
```
Microsoft (R) Build Engine version 16.1.65-preview+gc86abee47b for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 929.46 ms for F:\AppModel\repro\dotnet-core-target-manifest-bug-master\TargetManifestBug.csproj.
C:\Program Files\dotnet\sdk\3.0.100-preview6-011540\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.ComposeStore.targets(347,5): error MSB3024: Could not copy the file "F:\AppModel\repro\dotnet-core-target-manifest-bug-master\obj\Debug\netcoreapp2.2\win7-x64\TargetManifestBug.exe" to the destination file "C:\Users\vitkaras\.dotnet\store\x64\netcoreapp2.2\", because the destination is a folder instead of a file. To copy the source file into a folder, consider using the DestinationFolder parameter instead of DestinationFiles. [F:\AppModel\repro\dotnet-core-target-manifest-bug-master\TargetManifestBug.csproj]
  Files were composed in C:\Users\vitkaras\.dotnet\store\x64\netcoreapp2.2\
  The list of packages stored is in C:\Users\vitkaras\.dotnet\store\x64\netcoreapp2.2\artifact.xml
```

Running the same on .NET Core SDK 2.2.100 (using `global.json`) works just fine.
Seems like a regression in 3.0.

--- a/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ComposeStore.targets
+++ b/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ComposeStore.targets
@@ -101,7 +101,7 @@ Copyright (c) .NET Foundation. All rights reserved.
 <!-- Resolve phase-->
     <MSBuild Projects="@(PackageReferencesToStore)"
                  Targets="StoreResolver"
-                 Properties="SelfContained=false;MSBuildProjectExtensionsPath=$(ComposeWorkingDir)\%(PackageReferencesToStore.PackageName)_$([System.String]::Copy('%(PackageReferencesToStore.PackageVersion)').Replace('*','-'))\;"
+                 Properties="SelfContained=false;UseAppHost=false;MSBuildProjectExtensionsPath=$(ComposeWorkingDir)\%(PackageReferencesToStore.PackageName)_$([System.String]::Copy('%(PackageReferencesToStore.PackageVersion)').Replace('*','-'))\;"
                  BuildInParallel="$(BuildInParallel)">
       <Output ItemName="ResolvedPackagesFromMapper" TaskParameter="TargetOutputs" />
     </MSBuild>

--- a/src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToStoreAProjectWithDependencies.cs
+++ b/src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToStoreAProjectWithDependencies.cs
@@ -17,7 +17,7 @@ using Microsoft.NET.TestFramework.ProjectConstruction;
 
 using NuGet.Packaging.Core;
 using NuGet.Versioning;
-
+using Xunit;
 using Xunit.Abstractions;
 
 namespace Microsoft.NET.Publish.Tests
@@ -300,8 +300,10 @@ namespace Microsoft.NET.Publish.Tests
             }
         }
 
-        [CoreMSBuildOnlyFact]
-        public void It_stores_when_targeting_netcoreapp3()
+        [CoreMSBuildOnlyTheory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void It_stores_when_targeting_netcoreapp3(bool isExe)
         {
             const string TFM = "netcoreapp3.0";
 
@@ -309,12 +311,13 @@ namespace Microsoft.NET.Publish.Tests
             {
                 Name = "Test",
                 IsSdkProject = true,
-                TargetFrameworks = TFM
+                TargetFrameworks = TFM,
+                IsExe = isExe,
             };
 
             testProject.PackageReferences.Add(new TestPackageReference("Newtonsoft.Json", "12.0.1"));
 
-            var testProjectInstance = _testAssetsManager.CreateTestProject(testProject);
+            var testProjectInstance = _testAssetsManager.CreateTestProject(testProject, identifier: isExe.ToString());
 
             var outputFolder = Path.Combine(testProjectInstance.TestRoot, "o");
             var workingDir = Path.Combine(testProjectInstance.TestRoot, "w");

